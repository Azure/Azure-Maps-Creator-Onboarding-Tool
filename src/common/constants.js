export const PATHS = {
  INDEX: '/',
  VIEW_CONVERSIONS: '/view-conversions',
  CREATE_UPLOAD: '/create-upload',
  PROCESSING: '/processing',
  CREATE_GEOREFERENCE: '/create-georeference',
  REVIEW_CREATE: '/review-create',
  CONVERSION: '/review-create/conversion',
  IMDF_CONVERSION: '/review-create/imdf-conversion',
  PAST_CONVERSION: '/past-conversion',
  CONVERSIONS: '/conversions',
  LAYERS: '/layers',
  LEVELS: '/levels',
  INVALID_PATH: '/*',
};

export const ROUTE_NAME_BY_PATH = {
  [PATHS.INDEX]: 'home',
  [PATHS.CREATE_UPLOAD]: 'Create',
  [PATHS.VIEW_CONVERSIONS]: 'View',
  [PATHS.PROCESSING]: 'processing',
  [PATHS.CREATE_GEOREFERENCE]: 'prepare.drawing.package',
  [PATHS.LAYERS]: 'prepare.drawing.package',
  [PATHS.LEVELS]: 'prepare.drawing.package',
  [PATHS.INVALID_PATH]: 'redirect',
  [PATHS.REVIEW_CREATE]: 'prepare.drawing.package',
  [PATHS.CONVERSION]: 'create.indoor.map',
  [PATHS.IMDF_CONVERSION]: 'convert.indoor.map',
  [PATHS.CONVERSIONS]: 'All conversions',
};

export const PLACES_ROUTE_NAME_BY_PATH = {
  [PATHS.CREATE_GEOREFERENCE]: 'prepare.building.configuration',
  [PATHS.LAYERS]: 'prepare.building.configuration',
  [PATHS.LEVELS]: 'prepare.building.configuration',
  [PATHS.REVIEW_CREATE]: 'prepare.building.configuration',
};

export const TRUNCATE_FRACTION_DIGITS = 8;

export const HTTP_STATUS_CODE = {
  ACCEPTED: 202,
  OK: 200,
  BAD_REQUEST: 400,
  UNAUTHORIZED: 401,
  FORBIDDEN: 403,
  TOO_MANY_REQUESTS: 429,
};

export const isStagingEnv = process.env.REACT_APP_STAGING_ENV === 'true';

export const PLACES_PREVIEW = {
  DEFAULT_LANGUAGE: 'en',
  VERSION: 'places-1.0',
  DESCRIPTION: 'Converted by AzMapsCreatorOnboardingTool',
  STORAGE_RETENTION: 'day',
  SEARCH_PARAMETER: 'placespreview',
};
