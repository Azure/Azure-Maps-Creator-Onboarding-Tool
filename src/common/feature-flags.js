const flags = {
  onboardingEnabled: false,
};

export default flags;