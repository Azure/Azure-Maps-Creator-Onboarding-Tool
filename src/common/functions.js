/**
 * Given a single path name, returns a list of all path names from root to the given path name.
 * Ex. Given '/create/hello', returns ['/', '/create', '/create/hello']
 * @param {String} pathName the path name of the URL
 * @returns {String[]} an array of path names
 */
export const getSplitPaths = (pathName) => {
  if (typeof pathName !== 'string' || pathName === '') { return []; }
  if (pathName === '/') { return ['/']; }

  const pathNameLowerCase = pathName.toLowerCase();
  const splits = pathNameLowerCase.split('/');
  const splitPaths = [];
  for (let i = splits.length - 1; i > 0; i--) {
    let stringBuilder = '';
    for (let j = 1; j <= i; j++) {
      stringBuilder += '/' + splits[j];
    }
    splitPaths.push(stringBuilder);
  }
  splitPaths.push('/');
  splitPaths.reverse();

  return splitPaths;
};