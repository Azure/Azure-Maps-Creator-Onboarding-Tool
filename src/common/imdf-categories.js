export const imdfCategories = [
  'auditorium',
  'brick',
  'classroom',
  'column',
  'concrete',
  'conferenceroom',
  'drywall',
  'elevator',
  'escalator',
  'fieldofplay',
  'firstaid',
  'fitnessroom',
  'foodservice',
  'footbridge',
  'glass',
  'huddleroom',
  'kitchen',
  'laboratory',
  'library',
  'lobby',
  'lounge',
  'mailroom',
  'mothersroom',
  'movietheater',
  'movingwalkway',
  'nonpublic',
  'office',
  'opentobelow',
  'parking',
  'phoneroom',
  'platform',
  'privatelounge',
  'ramp',
  'recreation',
  'restroom',
  'restroom.family',
  'restroom.female',
  'restroom.female.wheelchair',
  'restroom.male',
  'restroom.male.wheelchair',
  'restroom.transgender',
  'restroom.transgender.wheelchair',
  'restroom.unisex',
  'restroom.unisex.wheelchair',
  'restroom.wheelchair',
  'road',
  'room',
  'serverroom',
  'shower',
  'smokingarea',
  'stairs',
  'steps',
  'storage',
  'structure',
  'terrace',
  'theater',
  'unenclosedarea',
  'unspecified',
  'vegetation',
  'waitingroom',
  'walkway',
  'walkway.island',
  'wood',
];
