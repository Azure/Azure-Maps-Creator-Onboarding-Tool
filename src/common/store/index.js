export { conversionSteps, useConversionStore } from './conversion.store';
export { useDissolvedExterior, useGeometryStore } from './geometry.store';
export { useLayersStore } from './layers.store';
export { useLevelsStore } from './levels.store';
export {
  progressBarStepsByKey,
  useCompletedSteps,
  useProgressBarSteps,
  useProgressBarStore,
} from './progress-bar-steps';
export { resetStores } from './reset';
export { LRO_STATUS, useResponseStore } from './response.store';
export { useReviewManifestJson, useReviewManifestStore } from './review-manifest.store';
export { getDomain, useUserStore } from './user.store';
