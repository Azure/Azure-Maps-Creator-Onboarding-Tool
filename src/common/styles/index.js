import { color, fontSize, fontWeight } from './styles';

export { color, fontSize, fontWeight };
