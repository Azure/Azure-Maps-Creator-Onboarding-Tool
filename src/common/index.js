import { PATHS, ROUTE_NAME_BY_PATH } from './constants';

export { PATHS, ROUTE_NAME_BY_PATH };