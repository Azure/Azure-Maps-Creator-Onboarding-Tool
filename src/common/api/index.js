// The following API is designed to be used only by the onboarding tool and is not supported for any other use.
import { getEnvs } from 'common/functions';
import { useUserStore } from '../store/user.store';

export const uploadFile = (file) => {
  const { geography, subscriptionKey } = useUserStore.getState();
  const url = `${getEnvs()[geography].URL}/manifest?api-version=2.0&subscription-key=${subscriptionKey}`;

  return fetch(url, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/zip',
    },
    body: file,
  });
};

export const fetchFromLocation = (location) => {
  const { subscriptionKey } = useUserStore.getState();
  return fetch(`${location}&subscription-key=${subscriptionKey}`, {
    method: 'GET',
  });
};

export const deleteFromLocation = (fetchUrl) => {
  const { subscriptionKey } = useUserStore.getState();
  fetch(`${fetchUrl}&subscription-key=${subscriptionKey}`, {
    method: 'DELETE',
  });
};

export const fetchAddress = (address) => {
  const { geography, subscriptionKey } = useUserStore.getState();
  return fetch(`${getEnvs()[geography].URL}/search/address/json?subscription-key=${subscriptionKey}&api-version=1.0&query=${address}&limit=1`)
    .then((res) => res.json());
};