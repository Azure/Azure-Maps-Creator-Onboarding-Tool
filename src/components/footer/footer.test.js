import { fireEvent, render, screen } from '@testing-library/react';
import { PATHS } from 'common';
import featureFlags from 'common/feature-flags';
import {
  useConversionStore,
  useGeometryStore,
  useLayersStore,
  useLevelsStore,
  useReviewManifestStore,
} from 'common/store';
import { saveAs } from 'file-saver';
import flushPromises from 'flush-promises';
import Footer, { TEST_ID } from './footer';

const mockNavigate = jest.fn();
let mockCurrentPathname = '/screen1';

jest.mock('hooks', () => ({
  useCustomNavigate: () => mockNavigate,
  useFeatureFlags: () => ({ isPlacesPreview: false }),
}));
jest.mock('react-router-dom', () => ({
  useLocation: () => ({
    pathname: mockCurrentPathname,
  }),
}));
jest.mock('common/feature-flags', () => ({
  onboardingEnabled: true,
}));

describe('footer on pages where it should not be rendered', () => {
  beforeEach(() => {
    mockCurrentPathname = '/a-snail-breathes-through-its-foot';
  });
  afterEach(() => {
    jest.resetAllMocks();
  });

  it('should render nothing', () => {
    const view = render(<Footer />);
    expect(view).toMatchSnapshot();
  });
});

describe('footer on pages where it should be rendered', () => {
  beforeEach(() => {
    mockCurrentPathname = PATHS.CREATE_GEOREFERENCE;
  });

  it('should render footer with next button disabled', () => {
    const view = render(<Footer />);
    expect(view).toMatchSnapshot();
  });

  it('should render footer with prev button disabled', () => {
    mockCurrentPathname = PATHS.LEVELS;
    useGeometryStore.setState({ checkedByUser: true });
    useLayersStore.setState({ layers: [{ id: 0, value: ['OUTLINE'], props: [] }] });
    useLevelsStore.setState({ levels: [{ levelName: '1', ordinal: 1 }] });
    const view = render(<Footer />);
    expect(view).toMatchSnapshot();
  });

  it('should navigate to next page when next button is clicked', () => {
    mockCurrentPathname = PATHS.LAYERS;
    useLayersStore.setState({ layers: [{ id: 0, value: ['OUTLINE'], props: [] }] });
    render(<Footer />);
    const nextBtn = screen.getByTestId(TEST_ID.NEXT_BUTTON);
    fireEvent.click(nextBtn);
    expect(mockNavigate).toHaveBeenCalledWith(PATHS.REVIEW_CREATE);
  });

  it('should navigate to prev page when prev button is clicked', () => {
    mockCurrentPathname = PATHS.LAYERS;
    render(<Footer />);
    const prevBtn = screen.getByTestId(TEST_ID.PREVIOUS_BUTTON);
    fireEvent.click(prevBtn);
    expect(mockNavigate).toHaveBeenCalledWith(PATHS.CREATE_GEOREFERENCE);
  });

  it('should navigate to review page when review+download is clicked', async () => {
    mockCurrentPathname = PATHS.LEVELS;
    render(<Footer />);

    const createDownloadBtn = screen.getByTestId(TEST_ID.CONVERT_BUTTON);
    fireEvent.click(createDownloadBtn);
    await flushPromises();

    expect(mockNavigate).toHaveBeenCalledWith(PATHS.REVIEW_CREATE);
  });

  it('should navigate to conversion page when create+download is clicked', async () => {
    mockCurrentPathname = PATHS.REVIEW_CREATE;
    useReviewManifestStore.setState({
      manifestReviewed: true,
    });
    const conversionState = useConversionStore.getState();
    const resetSpy = jest.spyOn(conversionState, 'reset');
    const uploadPackageSpy = jest.spyOn(conversionState, 'uploadPackage').mockImplementation(() => {});
    useGeometryStore.setState({ dwgLayers: ['layer1'] });
    useLayersStore.setState({ visited: true, layers: [] });
    useLevelsStore.setState({ levels: [{ levelName: '1', ordinal: '1', verticalExtent: '50' }] });

    render(<Footer />);
    const createDownloadBtn = screen.getByTestId(TEST_ID.CONVERT_BUTTON);
    fireEvent.click(createDownloadBtn);
    await flushPromises();

    expect(resetSpy).toHaveBeenCalled();
    expect(uploadPackageSpy).toHaveBeenCalled();
    expect(saveAs).toHaveBeenCalled();
    expect(mockNavigate).toHaveBeenCalledWith(PATHS.CONVERSION);
  });

  it('should not navigate to conversion page when onboarding FF is disabled', async () => {
    mockCurrentPathname = PATHS.REVIEW_CREATE;
    useReviewManifestStore.setState({
      manifestReviewed: true,
    });
    featureFlags.onboardingEnabled = false;
    const conversionState = useConversionStore.getState();
    const resetSpy = jest.spyOn(conversionState, 'reset');
    const uploadPackageSpy = jest.spyOn(conversionState, 'uploadPackage').mockImplementation(() => {});
    useGeometryStore.setState({ dwgLayers: ['layer1'] });
    useLayersStore.setState({ visited: true, layers: [] });
    useLevelsStore.setState({ levels: [{ levelName: '1', ordinal: '1', verticalExtent: '50' }] });

    render(<Footer />);
    const createDownloadBtn = screen.getByTestId(TEST_ID.CONVERT_BUTTON);
    fireEvent.click(createDownloadBtn);
    await flushPromises();

    expect(resetSpy).toHaveBeenCalled();
    expect(uploadPackageSpy).not.toHaveBeenCalled();
    expect(saveAs).toHaveBeenCalled();
    expect(mockNavigate).not.toHaveBeenCalledWith(PATHS.CONVERSION);
  });
});
