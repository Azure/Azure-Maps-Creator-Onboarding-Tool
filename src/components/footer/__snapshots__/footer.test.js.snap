// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`footer on pages where it should be rendered should render footer with next button disabled 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  position: fixed;
  width: 100%;
  bottom: 0;
  border-top: 1px solid #ccc;
  background-color: white;
  padding: 1rem 1.25rem;
}

.emotion-1 {
  margin-right: 0.625rem;
  height: 1.5rem;
  font-size: 0.813rem;
}

<body>
    <div>
      <div
        class="emotion-0"
      >
        <button
          class="ms-Button ms-Button--primary emotion-1 root-109"
          data-is-focusable="true"
          data-testid="convert-button"
          type="button"
        >
          <span
            class="ms-Button-flexContainer flexContainer-110"
            data-automationid="splitbuttonprimary"
          >
            review.plus.create
          </span>
        </button>
        <button
          class="ms-Button ms-Button--default emotion-1 root-117"
          data-is-focusable="true"
          data-testid="previous-button"
          type="button"
        >
          <span
            class="ms-Button-flexContainer flexContainer-110"
            data-automationid="splitbuttonprimary"
          >
            <span
              class="ms-Button-textContainer textContainer-111"
            >
              <span
                class="ms-Button-label label-113"
                id="id__3"
              >
                previous
              </span>
            </span>
          </span>
        </button>
        <button
          class="ms-Button ms-Button--default emotion-1 root-117"
          data-is-focusable="true"
          data-testid="next-button"
          type="button"
        >
          <span
            class="ms-Button-flexContainer flexContainer-110"
            data-automationid="splitbuttonprimary"
          >
            <span
              class="ms-Button-textContainer textContainer-111"
            >
              <span
                class="ms-Button-label label-113"
                id="id__6"
              >
                next
              </span>
            </span>
          </span>
        </button>
      </div>
    </div>
  </body>,
  "container": .emotion-0 {
  position: fixed;
  width: 100%;
  bottom: 0;
  border-top: 1px solid #ccc;
  background-color: white;
  padding: 1rem 1.25rem;
}

.emotion-1 {
  margin-right: 0.625rem;
  height: 1.5rem;
  font-size: 0.813rem;
}

<div>
    <div
      class="emotion-0"
    >
      <button
        class="ms-Button ms-Button--primary emotion-1 root-109"
        data-is-focusable="true"
        data-testid="convert-button"
        type="button"
      >
        <span
          class="ms-Button-flexContainer flexContainer-110"
          data-automationid="splitbuttonprimary"
        >
          review.plus.create
        </span>
      </button>
      <button
        class="ms-Button ms-Button--default emotion-1 root-117"
        data-is-focusable="true"
        data-testid="previous-button"
        type="button"
      >
        <span
          class="ms-Button-flexContainer flexContainer-110"
          data-automationid="splitbuttonprimary"
        >
          <span
            class="ms-Button-textContainer textContainer-111"
          >
            <span
              class="ms-Button-label label-113"
              id="id__3"
            >
              previous
            </span>
          </span>
        </span>
      </button>
      <button
        class="ms-Button ms-Button--default emotion-1 root-117"
        data-is-focusable="true"
        data-testid="next-button"
        type="button"
      >
        <span
          class="ms-Button-flexContainer flexContainer-110"
          data-automationid="splitbuttonprimary"
        >
          <span
            class="ms-Button-textContainer textContainer-111"
          >
            <span
              class="ms-Button-label label-113"
              id="id__6"
            >
              next
            </span>
          </span>
        </span>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`footer on pages where it should be rendered should render footer with prev button disabled 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  position: fixed;
  width: 100%;
  bottom: 0;
  border-top: 1px solid #ccc;
  background-color: white;
  padding: 1rem 1.25rem;
}

.emotion-1 {
  margin-right: 0.625rem;
  height: 1.5rem;
  font-size: 0.813rem;
}

<body>
    <div>
      <div
        class="emotion-0"
      >
        <button
          class="ms-Button ms-Button--primary emotion-1 root-109"
          data-is-focusable="true"
          data-testid="convert-button"
          type="button"
        >
          <span
            class="ms-Button-flexContainer flexContainer-110"
            data-automationid="splitbuttonprimary"
          >
            review.plus.create
          </span>
        </button>
        <button
          aria-disabled="true"
          class="ms-Button ms-Button--default is-disabled emotion-1 root-118"
          data-is-focusable="false"
          data-testid="previous-button"
          disabled=""
          type="button"
        >
          <span
            class="ms-Button-flexContainer flexContainer-110"
            data-automationid="splitbuttonprimary"
          >
            <span
              class="ms-Button-textContainer textContainer-111"
            >
              <span
                class="ms-Button-label label-113"
                id="id__12"
              >
                previous
              </span>
            </span>
          </span>
        </button>
        <button
          class="ms-Button ms-Button--default emotion-1 root-117"
          data-is-focusable="true"
          data-testid="next-button"
          type="button"
        >
          <span
            class="ms-Button-flexContainer flexContainer-110"
            data-automationid="splitbuttonprimary"
          >
            <span
              class="ms-Button-textContainer textContainer-111"
            >
              <span
                class="ms-Button-label label-113"
                id="id__15"
              >
                next
              </span>
            </span>
          </span>
        </button>
      </div>
    </div>
  </body>,
  "container": .emotion-0 {
  position: fixed;
  width: 100%;
  bottom: 0;
  border-top: 1px solid #ccc;
  background-color: white;
  padding: 1rem 1.25rem;
}

.emotion-1 {
  margin-right: 0.625rem;
  height: 1.5rem;
  font-size: 0.813rem;
}

<div>
    <div
      class="emotion-0"
    >
      <button
        class="ms-Button ms-Button--primary emotion-1 root-109"
        data-is-focusable="true"
        data-testid="convert-button"
        type="button"
      >
        <span
          class="ms-Button-flexContainer flexContainer-110"
          data-automationid="splitbuttonprimary"
        >
          review.plus.create
        </span>
      </button>
      <button
        aria-disabled="true"
        class="ms-Button ms-Button--default is-disabled emotion-1 root-118"
        data-is-focusable="false"
        data-testid="previous-button"
        disabled=""
        type="button"
      >
        <span
          class="ms-Button-flexContainer flexContainer-110"
          data-automationid="splitbuttonprimary"
        >
          <span
            class="ms-Button-textContainer textContainer-111"
          >
            <span
              class="ms-Button-label label-113"
              id="id__12"
            >
              previous
            </span>
          </span>
        </span>
      </button>
      <button
        class="ms-Button ms-Button--default emotion-1 root-117"
        data-is-focusable="true"
        data-testid="next-button"
        type="button"
      >
        <span
          class="ms-Button-flexContainer flexContainer-110"
          data-automationid="splitbuttonprimary"
        >
          <span
            class="ms-Button-textContainer textContainer-111"
          >
            <span
              class="ms-Button-label label-113"
              id="id__15"
            >
              next
            </span>
          </span>
        </span>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`footer on pages where it should not be rendered should render nothing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
